ouroboros-network-framework/test/Test/Ouroboros/Network/Orphans.hs

```diff
@@ -1,14 +1,46 @@
 {-# OPTIONS_GHC -Wno-orphans     #-}
+{-# OPTIONS_GHC -Wno-deferred-out-of-scope-variables #-}
 module Test.Ouroboros.Network.Orphans () where
 
+import           Data.List.NonEmpty (NonEmpty(..))
+
 import           Network.TypedProtocol.PingPong.Type (PingPong)
 import           Network.TypedProtocol.ReqResp.Type  (ReqResp)
 
 import           Ouroboros.Network.Util.ShowProxy (ShowProxy (..))
 
+import           Test.QuickCheck (Arbitrary (..), sized, Gen, choose, vectorOf)
+
+import           Simulation.Network.Snocket (Script (..))
+import qualified Data.List.NonEmpty as NonEmpty
+
 
 instance ShowProxy PingPong where
     showProxy _ = "PingPong"
 
 instance ShowProxy (ReqResp req resp) where
     showProxy _ = "ReqResp"
+
+instance Arbitrary a => Arbitrary (Script a) where
+    arbitrary = sized $ \sz -> arbitraryScriptOf sz arbitrary
+
+    shrink (Script (x :| [])) = [ Script (x' :| []) | x' <- shrink x ]
+    shrink (Script (x :| xs)) =
+        Script (x :| [])                          -- drop whole tail
+      : Script (x :| take (length xs `div` 2) xs) -- drop half the tail
+      : Script (x :| init xs)                     -- drop only last
+
+        -- drop none, shrink only elements
+      : [ Script (x' :| xs) | x'  <- shrink x ]
+     ++ [ Script (x :| xs') | xs' <- shrinkListElems shrink xs ]
```

Comment:
What's the advantage over `shrinkList`?

Reply:
I think I just copied this definition from the one in `ouroboros-network` didn't gave it much thought.